app/services/prompt_context.py

```
@@ -0,0 +1,280 @@
+        or parsed.is_unspecified
+    )
+
+
+def _validate_fetch_url(url: str) -> None:
+    """Block obvious SSRF targets unless explicitly allowed."""
+    parsed = urlparse(url)
+    if parsed.scheme not in {"http", "https"} or not parsed.hostname:
+        raise PromptContextError(
+            f"Invalid prompt data source URL: {url}",
+            status_code=400,
+        )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)